im-sdk-core/src/main/java/com/easemob/im/server/model/EMUser.java

```diff
@@ -63,18 +76,19 @@ public boolean equals(Object o) {
             return false;
         }
         EMUser emUser = (EMUser) o;
-        return username.equals(emUser.username);
+        return getUsername().equals(emUser.getUsername()) && getUuid().equals(emUser.getUuid());
```

Comment:
Don't include uuid in equals and hashCode.
BTW. The uuid is nullable.

Reply:
done